builder/amazon/ebssurrogate/builder.go

```diff
@@ -210,7 +209,6 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 			EbsOptimized:                      b.config.EbsOptimized,
 			EnableT2Unlimited:                 b.config.EnableT2Unlimited,
 			ExpectedRootDevice:                "ebs",
-			IamInstanceProfile:                b.config.IamInstanceProfile,
 			InstanceInitiatedShutdownBehavior: b.config.InstanceInitiatedShutdownBehavior,
 			InstanceType:                      b.config.InstanceType,
```

Comment:
I know this step doesn't make a lot of sense for the ebssurrogate builder, but there's a chance that not running it will cause issues with shared steps that expect to be able to read either an instance profile or "" from state -- I'd add the step just in case to make sure that the state is set up the same way for ebesurrogate as for the other builders.

Reply:
done, that certainly makes sense to keep them consistent